### PR TITLE
[12.0][FWPORT] mrp_multi_level: origin for "mv" was never being set.

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
Also, improve mrp.move by using picking name if the move is part of a picking.

Forward port of https://github.com/OCA/manufacture/pull/450.

@ForgeFlow

### CODE REVIEW

File: mrp_multi_level/wizards/mrp_multi_level.py [hunk 1]
```diff
@@ -55,25 +55,27 @@ def _prepare_mrp_move_data_from_stock_move(
...
+            for move_dest_id in move.move_dest_ids:
+                if move_dest_id.production_id:
+                    order_number = move_dest_id.production_id.name
+                    origin = 'mo'
+                    mo = move_dest_id.production_id.id
+                    if move_dest_id.production_id.product_id:
+                        parent_product_id = \
+                            move_dest_id.production_id.product_id.id
+                    else:
+                        parent_product_id = move_dest_id.product_id.id
```
Comment (MiquelRForgeFlow) [inline mrp_multi_level/wizards/mrp_multi_level.py]:
```suggestion
                    parent_product_id = (move_dest_id.production_id.product_id or move_dest_id.product_id).id
```

File: mrp_multi_level/wizards/mrp_multi_level.py [hunk 2]
```diff
@@ -55,25 +55,27 @@ def _prepare_mrp_move_data_from_stock_move(
...
-                            parent_product_id = \
-                                move_dest_id.production_id.product_id.id
-                        else:
-                            parent_product_id = move_dest_id.product_id.id
-        if order_number is None:
-            order_number = move.name
+            if move.picking_id:
+                order_number = move.picking_id.name
+            else:
+                order_number = move.name
```
Comment (MiquelRForgeFlow) [inline mrp_multi_level/wizards/mrp_multi_level.py]:
```suggestion
            order_number = (move.picking_id or move).name
```

**Review by MiquelRForgeFlow — APPROVED** [review]
code review :+1:

**Review by AdriaGForgeFlow — APPROVED** [review]
Code review :+1:

File: mrp_multi_level/wizards/mrp_multi_level.py [hunk 3]
```diff
@@ -55,25 +55,20 @@ def _prepare_mrp_move_data_from_stock_move(
...
             po = move.purchase_line_id.order_id.id
             po_line = move.purchase_line_id.id
-        if move.production_id:
+        elif move.production_id:
             order_number = move.production_id.name
             origin = 'mo'
             mo = move.production_id.id
+        elif move.move_dest_ids:
+            for move_dest_id in move.move_dest_ids:
+                if move_dest_id.production_id:
```
Comment (MiquelRForgeFlow) [inline mrp_multi_level/wizards/mrp_multi_level.py]: just asking: shouldn't this `if` have an `else` now?

Comment (LoisRForgeFlow) [reply mrp_multi_level/wizards/mrp_multi_level.py]: it has

Comment (MiquelRForgeFlow) [reply mrp_multi_level/wizards/mrp_multi_level.py]: where? 🤔

Comment (LoisRForgeFlow) [reply mrp_multi_level/wizards/mrp_multi_level.py]: ah, you mean "move_dest_id.production_id" no, it is not needed, it's uses de default origin "mv", that's the point of this fix.

Comment (MiquelRForgeFlow) [inline mrp_multi_level/wizards/mrp_multi_level.py]:
```suggestion
            for move_dest_id in move.move_dest_ids.filtered('production_id'):
```

Comment (MiquelRForgeFlow) [reply mrp_multi_level/wizards/mrp_multi_level.py]: ah, ok

Comment (LoisRForgeFlow) [reply mrp_multi_level/wizards/mrp_multi_level.py]: No more cosmetics fixes please :see_no_evil:

**Review by AaronHForgeFlow — APPROVED** [review]

**Pull request merged**